### PR TITLE
fix: email validation for braze

### PR DESCRIPTION
**Pull request by aashishmalik**
## What are the changes introduced in this PR?

email validation for braze if integer

Write a brief explainer on your code changes.

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Type of change

If the pull request is a **bug-fix**, **enhancement** or a **refactor**, please fill in the details on the changes made.

- Existing capabilities/behavior

- New capabilities/behavior

If the pull request is a **new feature**,

### Any technical or performance related pointers to consider with the change?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### If the PR has changes in more than 10 files, please mention why the changes were not split into multiple PRs.

N/A

### If multiple linear tasks are associated with the PR changes, please elaborate on the reason:

N/A

<hr>

### Developer checklist

- [ ] **No breaking changes are being introduced.**

- [ ] Are all related docs linked with the PR?

- [ ] Are all changes manually tested?

- [ ] Does this change require any documentation changes?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: src/v0/destinations/braze/transform.js [hunk 1]
```diff
@@ -155,8 +155,8 @@ function getUserAttributesObject(message, mappingJson, destination) {
     let value = get(traits, mappingJson[destKey]);
     if (value || (value === null && reservedKeys.includes(destKey))) {
       // if email is not string remove it from attributes
-      if (destKey === 'email' && typeof value !== 'string') {
-        throw new InstrumentationError('Invalid email, email must be a valid string');
+      if (destKey === 'email' && Number.isInteger(value)) {
```
Comment (koladilip) [inline src/v0/destinations/braze/transform.js:158]:
```js
switch (destKey) {
        case 'gender':
          value = formatGender(value);
          break;
        case 'email':
          if (typeof value !== 'string') {
            value = value.toString().toLowerCase();
          } else if (value !== undefined && value !== null){
            throw new InstrumentationError('Invalid email, email must be a valid string');
          }
          break;
        default:
          break;
      }
```
What about this?

Comment (aashishmalik) [reply src/v0/destinations/braze/transform.js:158]: if v is null ..then our first check v!== 'string' will be true ...we will never go to else if condition

Comment (koladilip) [reply src/v0/destinations/braze/transform.js:158]: corrected now

Comment (aashishmalik) [reply src/v0/destinations/braze/transform.js]: updated

File: src/v0/destinations/braze/transform.js [hunk 2]
```diff
@@ -154,16 +154,19 @@ function getUserAttributesObject(message, mappingJson, destination) {
...
+      switch (destKey) {
+        case 'gender':
+          value = formatGender(value);
+          break;
+        case 'email':
+          if (typeof value === 'string') {
+            value = value.toLowerCase();
+          } else if (isDefinedAndNotNull(value)) {
+            throw new InstrumentationError('Invalid email, email must be a valid string');
+          }
```
Comment (utsabc) [inline src/v0/destinations/braze/transform.js:166]: Lets not throw error previously we were skipping it

File: test/__tests__/data/braze_output.json [hunk 3]
```diff
@@ -1038,5 +1038,76 @@
...
+  {
+    "body": {
+      "FORM": {},
+      "JSON": {
+        "attributes": [
+          {
+            "_update_existing_only": false,
+            "city": "Disney",
+            "country": "USA",
+            "email": null,
```
Comment (utsabc) [inline test/__tests__/data/braze_output.json:1051]: Email is still coming as null

Comment (aashishmalik) [reply test/__tests__/data/braze_output.json:1051]: we want to allow email as this is reserved keys and should pass as it is.

File: src/v0/destinations/braze/transform.js (hunk 2 above)
Comment (aashishmalik) [reply src/v0/destinations/braze/transform.js:166]: shouldn't we throw error if email is not string or null .... as that event be rejected by braze

File: test/__tests__/data/braze_output.json (hunk 3 above)
Comment (aashishmalik) [reply test/__tests__/data/braze_output.json:1051]: earlier code if (value || (value === null && reservedKeys.includes(destKey))) { ... is checking against reservedKeys ..even if it is null

File: src/v0/destinations/braze/transform.js (hunk 2 above)
Comment (utsabc) [reply src/v0/destinations/braze/transform.js:166]: In the previous code we were not adding in `data` basically attributes. Check once

Comment (aashishmalik) [reply src/v0/destinations/braze/transform.js:166]: is `email:12`, `email: {a:b}` are these are rejected by braze
 "errors": [
        {
            "type": "The value provided for the 'email' field is not a valid email.",
            "input_array": "attributes",
            "index": 0
        }

**Review by koladilip — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**